main.go

```diff
@@ -98,38 +133,134 @@ func getConfig(params AppConfigParams) ([]string, error) {
 		return nil, err
 	}
 
-	vars := getVars(string(configuration.Configuration))
+	return configuration.Configuration, nil
+}
+
+// getVars parses a config in env format and returns a slice of variable-value strings like "VAR=value" suitable to
+// supply to the Env attribute of the os/exec Cmd struct.
+func getVars(config []byte) ([]string, error) {
+	vars, err := godotenv.Parse(bytes.NewReader(config))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse configuration from AppConfig: %w", err)
+	}
 
 	fmt.Printf("read %d lines from AppConfig\n", len(vars))
 	if verbose {
 		fmt.Printf("vars: %s\n", vars)
 	}
 
-	return vars, nil
+	varSlice := make([]string, 0, len(vars))
+	for k, v := range vars {
+		varSlice = append(varSlice, k+"="+v)
+	}
+
+	return varSlice, nil
 }
 
-func getVars(config string) []string {
-	lines := strings.Split(config, "\n")
+// updateConfig looks in the config file for variables that should be updated from the local environment, swaps out
+// the value, and sends the new config file to AWS AppConfig
+func updateConfig(params AppConfigParams, cfgBytes []byte) error {
+	newCfg, err := replaceConfigValues(cfgBytes)
+	if err != nil {
+		return fmt.Errorf("failure replacing values: %w", err)
+	}
+
+	err = deployNewConfig(params, newCfg)
+	if err != nil {
+		return fmt.Errorf("failed to deploy config: %w", err)
+	}
 
-	var vars []string
-	for _, l := range lines {
-		if parsed := parseLine(l); parsed != "" {
-			vars = append(vars, parsed)
+	if verbose {
+		fmt.Printf("updated config: %s\n", newCfg)
+	}
+	return nil
+}
+
+// replaceConfigValues substitutes values from the local environment into designated variables in the config file
+func replaceConfigValues(cfg []byte) (string, error) {
+	localEnv := os.Environ()
+	envVars, err := godotenv.Parse(strings.NewReader(strings.Join(localEnv, "\n")))
+	if err != nil {
+		return "", fmt.Errorf("failed to parse environment variables using godotenv.Parse: %w", err)
+	}
+
+	scanner := bufio.NewScanner(bytes.NewReader(cfg))
+	var output strings.Builder
+	for scanner.Scan() {
+		line := scanner.Text()
+		for k, v := range envVars {
+			var err error
+			line, err = replaceLine(line, k, v)
+			if err != nil {
+				return "", err
+			}
 		}
+		output.WriteString(line + "\n")
+	}
+
+	if err = scanner.Err(); err != nil {
+		return "", fmt.Errorf("error reading input: %w", err)
+	}
+	return output.String(), nil
+}
+
+// replaceLine handles one line of the config file, replacing the variable value if marked for update
+func replaceLine(line, variable, newValue string) (string, error) {
+	if !strings.HasPrefix(line, variable) {
+		return line, nil
+	}
+
+	if !strings.Contains(line, "{update}") {
```

Comment:
Do you want to check somehow that the `{update}` is in the comment, rather than in the value? Just seems like an edge case worth asking about.

Reply:
My hope was that the braces make it unique enough that it wouldn't happen in a normal value. If you think double braces or some other type of bracketing would be better, I'm open to suggestions. But it's easy enough to move this check down a few lines and check the comment portion instead, so I'll do that.